sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/VertxAsyncHttpClient.java

```diff
@@ -96,30 +107,70 @@ public Mono<HttpResponse> send(HttpRequest request, Context context) {
                 }
             });
 
-            // TODO (alzimmer)
-            // For now Vertx will always use a buffered request until reliability issues when using streamin can be
-            // resolved.
-            Flux<ByteBuffer> requestBody = request.getBody();
-            if (requestBody == null) {
-                vertxHttpRequest.end();
-            } else {
-                if (progressReporter != null) {
-                    requestBody = requestBody.map(buffer -> {
-                        progressReporter.reportProgress(buffer.remaining());
-                        return buffer;
-                    });
-                }
-
-                FluxUtil.collectBytesFromNetworkResponse(requestBody, request.getHeaders())
-                    .subscribeOn(scheduler)
-                    .subscribe(bytes -> vertxHttpRequest.write(Buffer.buffer(Unpooled.wrappedBuffer(bytes))),
-                        sink::error, vertxHttpRequest::end);
-            }
+            sendBody(sink, request, progressReporter, vertxRequest);
         }));
     }
 
     @Override
     public HttpResponse sendSync(HttpRequest request, Context context) {
         return send(request, context).block();
     }
+
+    private void sendBody(MonoSink<HttpResponse> sink, HttpRequest azureRequest,
+        ProgressReporter progressReporter, HttpClientRequest vertxRequest) {
+        BinaryData body = azureRequest.getBodyAsBinaryData();
+        if (body == null) {
+            vertxRequest.send(result -> {
+                if (result.failed()) {
+                    sink.error(result.cause());
+                }
+            });
+            return;
+        }
+
+        BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+        if (bodyContent instanceof ByteArrayContent
+            || bodyContent instanceof StringContent
+            || bodyContent instanceof SerializableContent) {
```

Comment:
I've added ByteBufferContent as a standalone but ListByteBufferContent will still use the Flux implementation as it would require writing a `ReadStream` or write handler.